R/calc_views.R

```diff
@@ -123,73 +105,6 @@ set_view_lab <- function(views){
   set_view_window(views, param = 'xlab', na.value="")
 }
 
-
-set_view_lim <- function(views){
-  y.include <- c('y','y1','y0','ytop','ybottom')
-  x.include <- c('x','x1','x0','xleft','xright')
-  views <- set_view_window(views, param = 'xlim', na.value=NA, ignore='window', remove=FALSE)
-  views <- set_view_window(views, param = 'ylim', na.value=NA, ignore='window', remove=FALSE)
-  
-  data <- list(y=summarize_args(views, y.include, ignore=c('window','gs.config')), 
-               x=summarize_args(views, x.include, ignore=c('window','gs.config')))
-
-  axs <- list(yaxs=summarize_args(views, c('yaxs'), ignore=c('gs.config')),
-              xaxs=summarize_args(views, c('xaxs'), ignore=c('gs.config')))
-  
-  definedSides <- unlist(c_unname(views), recursive = FALSE)
-  definedSides <- unique(unname(unlist(definedSides[grep("side", names(definedSides))])))
-
-  for(param in c('y','x')){
-    for (i in names(data[[param]])){
-      n.i <- as.numeric(i)
-      lim.name <- paste0(param,'lim')
-      axs.name <- paste0(param, 'axs')
-      view.side <- get_view_side(views, as.numeric(i), param)
-      match.side <- as.character(views_with_side(views, view.side))
-      data.var <- c_unname(data[[param]][match.side])
-      
-      if(!is.null(data.var) && all(is.na(data.var))){
-        if((view.side %% 2) == 0){ #even
-          otherSide <- c(2,4)[c(2,4) %in% definedSides[definedSides != view.side]]
-        } else { #odd
-          otherSide <- c(1,3)[c(1,3) %in% definedSides[definedSides != view.side]]
-        }
-        data.var <- c_unname(data[[param]][otherSide])
-        if(all(is.na(data.var))){ #So do data total
-          data.var <- c(0,1)
-        }
-      }
-      
-      data.lim <- range(data.var[is.finite(data.var)])
-      usr.lim <- views[[n.i]][['window']][[lim.name]][1:2]
-      views[[n.i]][['window']][[lim.name]] <- data.lim
-      views[[n.i]][['window']][[lim.name]][!is.na(usr.lim)] <- usr.lim[!is.na(usr.lim)]
-    
-      
-      usr.axs <- axs[[axs.name]][[n.i]]
-      
-      if (any(!is.na(usr.axs)) && any(usr.axs == 'o')) {
```

Comment:
This is part that I didn't really get. @lindsaycarr can you help me think of how to pull this into the `sides` model?